lsp-mode.el

```diff
@@ -1877,6 +1897,101 @@ WORKSPACE is the workspace that contains the progress token."
                        (cons status global-mode-string))
                       (t (remove status global-mode-string))))))
 
+
+;; code actions modeline
+
+(defvar-local lsp--modeline-code-actions-string nil
+  "Holds the current code action string on modeline.")
+
+(declare-function all-the-icons-octicon "ext:all-the-icons")
+
+(defun lsp--modeline-code-actions-icon ()
+  "Build the icon for modeline code actions."
+  (if (featurep 'all-the-icons)
+      (all-the-icons-octicon "light-bulb"
+                             :face 'lsp-modeline-code-actions-face
+                             :v-adjust -0.0575)
+    (propertize "💡" 'face 'lsp-modeline-code-actions-face)))
+
+(defun lsp--modeline-build-code-actions-string (actions)
+  "Build the string to be presented on modeline for code ACTIONS."
+  (-let* ((icon (lsp--modeline-code-actions-icon))
+          (first-action-string (propertize (->> actions
+                                                lsp-seq-first
+                                                (gethash "title")
+                                                (replace-regexp-in-string "[\n\t ]+" " "))
+                                           'face 'lsp-modeline-code-actions-face))
+          (single-action? (= (length actions) 1))
+          (string (if single-action?
+                      (format " %s %s " icon first-action-string)
+                    (format " %s %s %s " icon first-action-string
+                            (propertize (format "(%d more)" (seq-length actions))
+                                        'display `((height 0.9))
+                                        'face 'lsp-modeline-code-actions-face)))))
+    (propertize string
+                'help-echo (concat "Apply code actions (s-l a a)\nmouse-1: "
+                                   (if single-action?
+                                       first-action-string
+                                     "select from multiple code actions"))
+                'mouse-face 'mode-line-highlight
+                'local-map (make-mode-line-mouse-map
+                            'mouse-1 (lambda ()
+                                       (interactive)
+                                       (if single-action?
+                                           (lsp-execute-code-action (lsp-seq-first actions))
+                                         (call-interactively #'lsp-execute-code-action)))))))
```

Comment:
Can we reuse the already calculated code actions?

Reply:
How? are they set in some variable already?

Reply:
I didn't find it anywhere

Reply:
aren't the actions present in the variable `actions`? `lsp-execute-code-action` will go to the server and calculate them again.

Reply:
Oh I see what do you mean, the problem is that `lsp-execute-code-action` receive the action that it will be executed or nothing to present to user select the desired action, currently there'is no way to pass actions to it

Reply:
``` (lsp-execute-code-action (lsp--select-action actions)) ```?

Reply:
Hum, I didn't notice we can do that, thanks!